src/main/kotlin/com/openlattice/transporter/types/TransporterDatastore.kt

```diff
@@ -0,0 +1,318 @@
+                    .split("\n")
+                    .forEach { sql ->
+                        logger.info("running {}", sql)
+                        st.execute(sql)
+                    }
+            conn.commit()
+        }
+    }
+
+    fun importTablesFromForeignSchema(
+            remoteSchema: String,
+            remoteTables: Set<String>,
+            localSchema: String,
+            usingFdwName: String
+    ): String {
+        val tablesClause = if ( remoteTables.isEmpty() ){
+            ""
+        } else {
+            "LIMIT TO (${remoteTables.joinToString(",")})"
+        }
+        return "import foreign schema $remoteSchema $tablesClause from server $usingFdwName INTO $localSchema;"
+    }
+
+    private fun ensureSearchPath( connection: Connection): String {
+        logger.info("checking search path for current user")
+        connection.createStatement().executeQuery( "show search_path" ).use {
+            it.next()
+            val searchPath = it.getString(1)
+            logger.info(searchPath)
+            if ( searchPath == null ) {
+                logger.error("bad search path: {}", searchPath)
+                return ""
+            }
+            return searchPath
+        }
+    }
+
+    private fun initializeFDW(rhizomeConfig: PostgresConfiguration) {
+        createFdwBetweenDatabases(
+                hds,
+                rhizomeConfig.hikariConfiguration.getProperty("username"),
+                rhizomeConfig.hikariConfiguration.getProperty("password"),
+                rhizomeConfig.hikariConfiguration.getProperty("jdbcUrl"),
+                assemblerConfiguration.server.getProperty("username"),
+                ENTERPRISE_FDW_SCHEMA,
+                ENTERPRISE_FDW_NAME
+        )
+
+        hds.connection.use { conn ->
+            conn.createStatement().use { stmt ->
+                stmt.executeQuery("select count(*) from information_schema.foreign_tables where foreign_table_schema = '$ENTERPRISE_FDW_SCHEMA'").use { rs ->
+                    if (rs.next() && rs.getInt(1) > 0) {
+                        // don't bother if it's already there
+                        logger.info("schema already imported, not re-importing")
+                    } else {
+                        stmt.executeUpdate(
+                                importTablesFromForeignSchema(
+                                        PUBLIC_SCHEMA,
+                                        setOf(
+                                                PostgresTable.IDS.name,
+                                                PostgresTable.DATA.name,
+                                                PostgresTable.E.name
+                                        ),
+                                        ENTERPRISE_FDW_SCHEMA,
+                                        ENTERPRISE_FDW_NAME
+                                )
+                        )
+                    }
+                }
+            }
+        }
+
+        hds.close()
```

Comment:
why is hds being closed here?

Reply:
(closed / set to transporter?)

Reply:
We want to re-create the datasource so that the cached connection has access to the foreign data wrapper. I wasn't seeing this in the documentation but I was witnessing it when testing locally

Reply:
huh interesting. yeah anyway this doesn't seem problematic, was just curious!

Reply:
might be worthy of a comment actually since I could see us down the road being like "why is this here" and removing it, and then things breaking